python/lsst/meas/extensions/piff/piffPsfDeterminer.py

```diff
@@ -32,6 +33,22 @@
 
 
 class PiffPsfDeterminerConfig(BasePsfDeterminerTask.ConfigClass):
+    def _validateGalsimIntepolant(name: str) -> bool:  # noqa: N805
+        """A helper function to validate the GalSim interpolant at config time.
+        """
+        # First, check if ``name`` is a valid Lanczos interpolant.
+        pattern = re.compile(r"Lanczos\(\d+\)")
```

Comment:
Better to let `GalSim` itself determine valid names here I think (to handle to off-chance that we add a new interpolant there, for example).  You can use `galsim.Interpolant.from_name()` and look for `GalSimValueError`.

Reply:
There's some difference in the way PiFF accepts interpolant names and GalSim does. For e.g., GalSim accepts `sinc` but PiFF needs it to be `galsim.SincInterpolant`. And `galsim.Interpolant.from_name('sinc').__class__` returns `galsim.Interpolant.SincInterpolant`, which is not accepted by PiFF. I don't foresee using anything other than `Lanczos`, and since the doc says we support only the ones listed there, if there are additional interpolation schemes that get added in GalSim, we can add that later. We should consider it as a technical debt at this point the very least.

Reply:
Okay.

Reply:
Or we could mock up what PiFF does currently. We can do
```python
import galsim
from galsim import Lanczos
...
def _validateGalSimInterpolant(name):
    try:
        eval(name)
        return True
    except AttributeError:
        return False
```

Reply:
Oh, interesting.  Looks like Piff discourages use of anything non-Lanczos?  Which is probably reasonable.  I like this best.

Reply:
On second thoughts, any executable code would pass this, even if it is not a valid interpolant name. And it'd fail later during runtime. Also, from a security standpoint, we don't want to execute an arbitrary code here.

Reply:
I guess you could add an `isinstance(blah, galsim.Interpolant)` check.  

As for security, I don't know.  GalSim and piff both contain `eval` statements and we use them without worry.  We store piff results in the stack as pickles and I'm pretty sure you can hide arbitrary instructions in those.  I kinda think the ship has sailed with regard to python eval.